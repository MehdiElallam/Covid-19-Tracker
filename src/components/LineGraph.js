import React, { useEffect, useState } from 'react'
import { Line } from "react-chartjs-2";
import Chart from 'chart.js/auto'
import { Row } from 'react-bootstrap'
import helpers from '../utils/helpers'

export default function LineGraph(props) {

  const [chartData, setChartData] = useState(
    {
      labels: helpers.monthList(),
      datasets: [
        chartOptions('First dose', 'red', []),
        chartOptions('Second dose', 'green', []),
        chartOptions('Third dose', 'blue', [])
      ]
    }
  )

  useEffect(() => {
      
      props.vaccination && setChartData({
        labels: helpers.monthList(),
        datasets: [
          chartOptions('First dose', '#77c4bf', DoseData(props.vaccination, 1)),
          chartOptions('Second dose', '#009c91', DoseData(props.vaccination, 2)),
          chartOptions('Third dose', '#00413d', DoseData(props.vaccination, 3))
        ]
      })
  
  }, [props.vaccination])
  

    return (
        <Row>
            <h4>{props.title}</h4>
            <hr />
            { props.vaccination && <Line data={chartData} /> }
        </Row>
    )
}

const DoseData = (data, dose) => {
  
  const graphData = []

  if(data){
    const currentDate = helpers.currentDate();
    const currentMonth = parseInt(helpers.currentDate('month'))
  
    for (let index = 1; index <= currentMonth; index++) {
      
      let thisMonth = helpers.dateByMonth(index);
      let firstDay = helpers.getFirstLastDay(thisMonth).firstDay
      let lastDay = index === currentMonth ? currentDate : helpers.getFirstLastDay(thisMonth).lastDay
      let Dose = dose === 1 ? (data[lastDay] - data[firstDay] ) / 2.5 : dose === 2 ? (data[lastDay] - data[firstDay] ) / 3 : (data[lastDay] - data[firstDay] ) / 6
      
      graphData.push(Dose)
      
    }
  }
  

  return graphData

}

const chartOptions = (label, color, data) =>{
  return {
    label,
    fill: false,
    lineTension: 0.1,
    backgroundColor: color,
    borderColor: color,
    data
  }
}