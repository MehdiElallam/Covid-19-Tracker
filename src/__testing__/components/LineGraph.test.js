import React from 'react';
import { configure, shallow } from 'enzyme';
import LineGraph from '../../components/LineGraph';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe("Rendring LineGraph component", () => {


    it('LineGraph should render correctly with no props', () => {
        
        const component = shallow(<LineGraph/>);
        expect(component).toMatchSnapshot();
    });

    it('LineGraph should render correctly with vaccination props equal to empty object', () => {
        
        const component = shallow(<LineGraph vaccination={{}} />);
        expect(component).toMatchSnapshot();
    });

});
  
  
// it('LineGraph should render correctly with given props', () => {
    
//     const component = shallow(<LineGraph title="Cases" number={110} />);
//     expect(component).toMatchSnapshot();
// });