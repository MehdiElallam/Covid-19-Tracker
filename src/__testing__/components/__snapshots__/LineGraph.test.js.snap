// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineGraph should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Rendring LineGraph component LineGraph should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Rendring LineGraph component LineGraph should render correctly with vaccination props equal to empty object 1`] = `ShallowWrapper {}`;
